plugins/faustwp/includes/replacement/callbacks.php

```diff
@@ -200,13 +200,39 @@ function post_preview_link( $link, $post ) {
  * @return string URL used for the post.
  */
 function post_link( $link ) {
+	global $pagenow;
+	$target_pages = array( 'admin-ajax.php', 'index.php', 'edit.php', 'post.php', 'post-new.php', 'upload.php', 'media-new.php' );
+
+	if ( empty( $_POST ) && 'post-new.php' === $pagenow ) {
+		return $link;
+	}
+
+	// Ajax requests to generate permalink.
+	if ( in_array( $pagenow, $target_pages, true )
+		&& ! empty( $_POST['samplepermalinknonce'] )
+		&& check_ajax_referer( 'samplepermalink', 'samplepermalinknonce' )
+	) {
+			return $link;
+	}
+
 	if (
 		! is_rewrites_enabled()
 		|| ( function_exists( 'is_graphql_request' ) && is_graphql_request() )
+		// Block editor makes REST requests on these pages to query content.
+		|| ( in_array( $pagenow, $target_pages, true ) && current_user_can( 'edit_posts' ) && defined( 'REST_REQUEST' ) && REST_REQUEST )
 	) {
 		return $link;
 	}
 
+	// Check for wp-link-ajax requests. Used by Classic Editor when linking content.
+	if ( wp_doing_ajax()
```

Comment:
This probably needs to be refactored as a function:

https://refactoring.guru/decompose-conditional

Reply:
Refactored in ca09f06279a12e4924829791f15752802ba037c5